plugins/outputs/wavefront/wavefront.go

```diff
@@ -173,7 +173,11 @@ func (w *Wavefront) Write(metrics []telegraf.Metric) error {
 		for _, point := range w.buildMetrics(m) {
 			err := w.sender.SendMetric(point.Metric, point.Value, point.Timestamp, point.Source, point.Tags)
 			if err != nil {
-				return fmt.Errorf("Wavefront sending error: %s", err.Error())
+				if isRetryable(err) {
+					return fmt.Errorf("Wavefront sending error: %v", err)
+				}
+				w.Log.Errorf("non-retryable error during Wavefront.Write: %v", err)
+				return nil
```

Comment:
You might not want to return here, and instead finish writing the batches, as some of those metrics are probably still good.

Reply:
Good point. Done.